Flask API: CA Business Entity endpoint error

test output:
```
~/maintest$ pytest -vvv
================================= test session starts =================================
platform linux -- Python 3.8.18, pytest-8.3.3, pluggy-1.5.0 -- /nix/store/dydlfqsd3al5d06h18922sp60b67pk32-python-wrapped-0.1.0/bin/python3
cachedir: .pytest_cache
rootdir: /home/runner/maintest
configfile: pyproject.toml
collected 6 items                                                                     

test_api.py::test_search_fda PASSED                                             [ 16%]
test_api.py::test_search_k510 PASSED                                            [ 33%]
test_api.py::test_search_cdph PASSED                                            [ 50%]
test_api.py::test_search_maude PASSED                                           [ 66%]
test_api.py::test_search_openhistorical PASSED                                  [ 83%]
test_api.py::test_search_cabusinessentity FAILED                                [100%]

====================================== FAILURES =======================================
____________________________ test_search_cabusinessentity _____________________________

    def test_search_cabusinessentity():
        response = requests.post("https://api.healthly.dev/ca-business-entity",
                                 json={"searchTerm": "heart"})
    
>       assert response.status_code == 200
E    assert 500 == 200
E     +  where 500 = <Response [500]>.status_code

test_api.py:53: AssertionError
=============================== short test summary info ===============================
FAILED test_api.py::test_search_cabusinessentity - assert 500 == 200
 +  where 500 = <Response [500]>.status_code
============================= 1 failed, 5 passed in 7.18s =============================
```
detailed error message:
```
INFO:werkzeug:172.18.0.1 - - [30/Oct/2024 22:25:55] "POST /openhistorical HTTP/1.1" 200 -
INFO:root:Received a CA business entity search request.
INFO:root:CA business entity request data: {'searchTerm': 'heart'}
INFO:root:Sending initial request to CA Secretary of State business search: https://bizfileonline.sos.ca.gov/search/business
INFO:root:Performing search with criteria: {'SearchType': 'CORP', 'SearchCriteria': 'heart', 'SearchSubType': 'Keyword'}
ERROR:root:Error fetching data from CA Secretary of State business search: 405 Client Error: Method Not Allowed for url: https://bizfileonline.sos.ca.gov/search/business
```

--- a/AwesomeProject/Backend/main.py
+++ b/AwesomeProject/Backend/main.py
@@ -360,7 +360,7 @@ def search_ca_business_entity():
 
         # Use the form data to perform the search
         logging.info(f"Performing search with criteria: {json_data}")
-        response = requests.post(search_url, data=json_data)
+        response = requests.post(search_url, json=json_data)
         response.raise_for_status()
 
         # Parse the search results

--- a/AwesomeProject/Backend/main.py
+++ b/AwesomeProject/Backend/main.py
@@ -357,10 +357,28 @@ def search_ca_business_entity():
             'SEARCH_VALUE': 'test',
             'SEARCH_TYPE_ID': '1',
         }
+        headers = {
+            'User-Agent':
+            'Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0',
+            'Accept': '*/*',
+            'Accept-Language': 'en-US,en;q=0.5',
+            'Referer': 'https://bizfileonline.sos.ca.gov/search/business',
+            'authorization': 'undefined',
+            'content-type': 'application/json',
+            'Origin': 'https://bizfileonline.sos.ca.gov',
+            'Sec-GPC': '1',
+            'Connection': 'keep-alive',
+            'Sec-Fetch-Dest': 'empty',
+            'Sec-Fetch-Mode': 'cors',
+            'Sec-Fetch-Site': 'same-origin',
+            'DNT': '1',
+            'Pragma': 'no-cache',
+            'Cache-Control': 'no-cache',
+        }
 
         # Use the form data to perform the search
         logging.info(f"Performing search with criteria: {json_data}")
-        response = requests.post(search_url, json=json_data)
+        response = requests.post(search_url, json=json_data, headers=headers)
         response.raise_for_status()
 
         # Parse the search results

--- a/AwesomeProject/Backend/main.py
+++ b/AwesomeProject/Backend/main.py
@@ -346,52 +346,36 @@ def search_ca_business_entity():
         return jsonify({"error": "Search term is required"}), 400
 
     # Build the search URL for the new data source
-    search_url = "https://bizfileonline.sos.ca.gov/search/business"
-
-    headers = {
-        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3',
-        'Accept-Language': 'en-US,en;q=0.9',
-        'Referer': 'https://bizfileonline.sos.ca.gov/',
-        'Origin': 'https://bizfileonline.sos.ca.gov'
-    }
+    search_url = "https://bizfileonline.sos.ca.gov/api/Records/businesssearch"
 
     try:
         # Perform the initial request to get the search page
         logging.info(f"Sending initial request to CA Secretary of State business search: {search_url}")
-        initial_response = requests.get(search_url, headers=headers)
-        initial_response.raise_for_status()
 
         # Parse the search page to get the necessary form data and cookies
-        soup = BeautifulSoup(initial_response.content, 'html.parser')
-        form_data = {
-            "SearchType": "CORP",
-            "SearchCriteria": search_term,
-            "SearchSubType": "Keyword"
+        json_data = {
+            'SEARCH_VALUE': 'test',
+            'SEARCH_TYPE_ID': '1',
         }
 
         # Use the form data to perform the search
-        logging.info(f"Performing search with criteria: {form_data}")
-        response = requests.post(search_url, data=form_data, headers=headers, cookies=initial_response.cookies)
+        logging.info(f"Performing search with criteria: {json_data}")
+        response = requests.post(search_url, data=json_data)
         response.raise_for_status()
 
         # Parse the search results
-        soup = BeautifulSoup(response.content, 'html.parser')
+        table_rows = response.json()["rows"]
         results = []
-
-        # Example scraping logic: Extract table rows containing the business entity data
-        table_rows = soup.select('table tbody tr')
-        for row in table_rows:
-            cells = row.find_all('td')
-            if len(cells) > 0:
-                result = {
-                    "entityInformation": cells[0].text.strip(),
-                    "initialFilingDate": cells[1].text.strip(),
-                    "status": cells[2].text.strip(),
-                    "entityType": cells[3].text.strip(),
-                    "formedIn": cells[4].text.strip(),
-                    "agent": cells[5].text.strip() if len(cells) > 5 else 'N/A'
-                }
-                results.append(result)
+        for k, v in table_rows.items():
+            result = {
+                "entityInformation": v["TITLE"][0],
+                "initialFilingDate": v["FILING_DATE"],
+                "status": v["STATUS"],
+                "entityType": v["ENTITY_TYPE"],
+                "formedIn": v["FORMED_IN"],
+                "agent": v["AGENT"]
+            }
+            results.append(result)
 
         return jsonify(results)
     except requests.RequestException as e:

